spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java

```diff
@@ -1181,7 +1170,7 @@ public synchronized void testDeleteWithSnapshotIsolation()
                   }
 
                   appendFiles.commit();
-                  sleep(10);
+                  Awaitility.await().pollInterval(Duration.ofMillis(10)).until(() -> true);
```

Comment:
I don't think this makes a lot of sense to replace this sleep with Awaitility. I wonder whether this sleep here is even necessary

Reply:
Maybe got the reason of the test failure. The reason seems to be the conflicts of the appended file names. When I tested using randomized names for each appendFile, all the tests succeeded. And in this case, as you said, `sleep` is no longer needed. Let me fix the file name configuration.